### PR TITLE
Fix warnings

### PR DESCRIPTION
**Pull request by Levminer**
As requested on stream :)

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -141,6 +141,9 @@
 //!   [sample]: https://gist.github.com/loderunner/36724cc9ee8db66db305#profiling-with-sample
 //!   [VTune]: https://software.intel.com/en-us/vtune-amplifier-help-command-line-interface
 
+// Remove when https://github.com/rust-lang/rust/issues/101730 lands
+#![allow(unknown_lints)]
```
Comment (jonhoo) [inline src/lib.rs:145]: Something seems weird here. After all, `rustdoc::all` shouldn't be enabling `rustdoc::missing_doc_code_examples` if that's not stable yet (if it does, that feels like a bug). Can we instead just remove the `#[allow]`?

Comment (Levminer) [reply src/lib.rs:145]: Its enabled on this line: https://github.com/Levminer/inferno/blob/3419a93cbb12c3a867011862f7d43f0e398f09b3/src/lib.rs#L148

Comment (jonhoo) [reply src/lib.rs:145]: That's an `allow()` line, so it's just making us _not_ get warnings about that particular lint.

Comment (Levminer) [reply src/lib.rs:145]: Ah, okay, but it DOES give warnings 😂. Nevermind, removed that line.

**Review by jonhoo — APPROVED** [review]
Thanks!

**Pull request merged**